Utilities/cond.cpp

```diff
@@ -12,6 +12,15 @@ bool cond_variable::imp_wait(u32 _old, u64 _timeout) noexcept
 	verify(HERE), _old != -1; // Very unlikely: it requires 2^32 distinct threads to wait simultaneously
 
 #ifdef _WIN32
+	if (_timeout > INTMAX_MAX / 10) // If it overflows, make it endless
+	{
+		_timeout = -1; // Won't ever finish anyway
+	}
+	else
+	{
+		_timeout *= -10;
```

Comment:
Why?

Reply:
10 because of the different time units between the winapi call and the ps3 one, negative means we send a relative timeout, and not an absolute time.

Reply:
@danilaml Nt API uses 100 nanosecond units not ms so we need the * 10 to convert. Negative means relative to time of submission, positive is absolute time.

Reply:
@kd-11 ok. SO is any positive number treated as infinite timeout then? Tried to search for docs but the best I've found didn't really go into params much.

Reply:
It allows you to set up absolute time as a trigger, e.g today at midnight or now + 10 minutes. Its not a timeout but a time point. There is a special infinite value to wait forever iirc

Reply:
@kd-11 hm, so to what absolute time does `10` translate exactly? What if it's before current time?

Reply:
10 would be 1ms after epoch which would be over 4 centuries ago. If timepoint < systemtime the timeout is ignored as it is already fulfilled, same as 0 timeout. Same thing happens if you move the system time forward beyond a future timepoint.